src/hotspot/cpu/ppc/macroAssembler_ppc.cpp

```diff
@@ -2723,13 +2723,37 @@ void MacroAssembler::compiler_fast_unlock_object(ConditionRegister flag, Registe
   b(success);
 
   bind(notRecursive);
-  ld(temp,             in_bytes(ObjectMonitor::EntryList_offset()), current_header);
-  ld(displaced_header, in_bytes(ObjectMonitor::cxq_offset()), current_header);
-  orr(temp, temp, displaced_header); // Will be 0 if both are 0.
-  cmpdi(flag, temp, 0);
-  bne(flag, failure);
+  const Register temp2 = displaced_header;
+
+  // Set owner to null.
   release();
+  li(temp, 0);
   std(temp, in_bytes(ObjectMonitor::owner_offset()), current_header);
+  // We need a full fence after clearing owner to avoid stranding.
+  // StoreLoad achieves this.
+  membar(StoreLoad);
```

Comment:
```suggestion
  fence();
```
similar to S390

Reply:
I use `membar(StoreLoad)` in all other platforms except in S390, and that is because it has no `membar()` implementation. I want to be consistent and choose to go with prior art, which happens to be the `OrderAccess::storeload()` called from `ObjectMonitor::exit()`. In PowerPC both `fence() `and `membar(StoreLoad)` are mapped to a `sync` instruction, so there is no real difference. But for consistency reasons I choose  `membar(StoreLoad)`. You know, "when in Rome"...